web/admin/application/library/IvozProvider/Klear/Ghost/RegisterStatus.php

```diff
@@ -360,4 +364,64 @@ private function getLocationStatus($model)
 
         return $registerStatus;
     }
+
+    /***
+     * @param $contact
+     * @param $received
+     * @return string
+     */
+    private function getRegisterHint($contact, $received) : string
+    {
+        $ctc = $this->isRFC1918($contact) ? 'private' : 'public';
+
+        if (is_null($received)) {
+            if ($ctc === 'private') {
+                $comment = "No NAT with private Contact (hint: internal routing)";
+            } else {
+                $comment = "No NAT with public Contact (hint: SIP ALG / STUN)";
+            }
+        } else {
+            $rcv = $this->isRFC1918($received) ? 'private' : 'public';
+
+            if ($ctc === 'private' && $rcv === 'public') {
+                $comment = "Regular NAT detected (private Contact, public Received)";
+            } else {
+                $comment = "Awkward NAT detected ($ctc Contact, $rcv Received)";
+            }
+        }
```

Comment:
what about this code 

```suggestion
        $contactIsPrivate = $this->isRFC1918($contact);

        if (is_null($received)) {
            if ($contactIsPrivate) {
                $comment = "No NAT with private Contact (hint: internal routing)";
            } else {
                $comment = "No NAT with public Contact (hint: SIP ALG / STUN)";
            }
        } else {
            $receivedIsPublic = $this->isRFC1918($received);

            if ($contactIsPrivate && $receivedIsPublic) {
                $comment = "Regular NAT detected (private Contact, public Received)";
            } else {
                $comment = "Awkward NAT detected (" 
                          . $contactIsPrivate ?  "Private" : "Public"
                          . "Contact, "
                          . $receivedIsPublic ?  "Public" : "Private"
                          . "Received)";
            }
        }
```

Use variables with more letters forming words in PHP



```

```